### ISSUE TITLE
Problems connecting LN wallets to BTCPayServer

### ISSUE THREAD

**Issue by vanjaime** [btcpayserver/btcpayserver-docker#345]
Hi, I am having problems connecting my zap wallet to BTCPayServer running over tor.

It looks the issue is related to no certificate for the gRPC config file.
![Screenshot_20200801-220125__01](https://user-images.githubusercontent.com/16761309/89113760-05988380-d443-11ea-861b-0cb689149fbe.jpg)

Note the certificate thumbprint 'null'
Shouldn't this have a cert?

![Screenshot_20200801-220209__01](https://user-images.githubusercontent.com/16761309/89113776-40022080-d443-11ea-9c15-a7bc5343da9e.jpg)

BTCPayServer deployed via Docker

**Comment by NicolasDorier** [comment]
@vanjaime BTCPay SErver is using a real certificate, a certificate is only needed if the secure channel is done through certificate pinning. That's not the case of BTCPay Server.

Seems you use Zap? It should work normally ping @rockstardev

**Comment by vanjaime** [comment]
That is correct I'm using ZAP however I tested Zeus as well and same issues connecting.
Here I cross-post a reply from ZAP github issues regarding the use of SSL gRPC and btcpayserver.
Thanks for looking into this.

**Comment by NicolasDorier** [comment]
@vanjaime Zeus is not using gRPC but REST so this is not the same QR Code. If Zeus does not work, it means LND is drunk and you need to restart it. `docker restart btcpayserver_lnd_bitcoin` or `btcpay-restart.sh` restart everything.

**Comment by vanjaime** [comment]
Sorry I meant I tested Zeus via REST but had similar issues. Looks like the server behind the hidden server is reached but the lack of SSL cert is what prevents ZAP from connecting.
See the config file generated where ssl field is false.

**Comment by NicolasDorier** [comment]
@vanjaime for Zap, I think you cna't use tor. Because they use gRPC which need Http2 which need HTTPS.

For Zeus, it should work, if you use Orbot.

**Comment by Eskyee** [comment]
Any updates or Info on connecting Zap LND over Tor yet   ??? as my iOS zap can't connect via the QR code to my BTCPay Hack0   :(
checked on Zap support, I found others still having this problem.  @rockstardev
